version.py

```diff
@@ -93,11 +92,48 @@
         prerel = "a"
     strictversion += prerel + str(version_info[-1])
 
-hexversion = version_info[4]
-hexversion |= RELEASE_LEVEL_VALUE.get(version_info[3], 0) * 1 << 4
-hexversion |= version_info[2] * 1 << 8
-hexversion |= version_info[1] * 1 << 16
-hexversion |= version_info[0] * 1 << 24
+_PATTERN = None
+
+
+def calc_hexversion(major=0, minor=0, micro=0, releaselevel="dev", serial=0, string=None):
+    """Calculate the hexadecimal version number from the tuple version_info:
+
+    :param major: integer
+    :param minor: integer
+    :param micro: integer
+    :param relev: integer or string
+    :param serial: integer
+    :return: integer always increasing with revision numbers
+    """
+    if string is not None:
```

Comment:
I wonder if 
try:
    parts = string.split(".", 2)
    parts[2], releaselevel = parts[2].split("-",1)
    major, minor, micro = (int(i) for i in parts)
except Exception as err:
    raise ValueError("'%s' is not a valid version" % string)

It is twice shorter (6 vs 12 lines), prevents the use of one global, one extra module, especially THE one responsible for so many regression in python3.7 (not to mention I don't  like regular expression)

Reply:
Your proposal do not match the version format as it is provided by fabio. Then i don't think it's a good idea.

I personally have no problems with the amount of lines, the extra standard module, nor the global as we use it everywhere. But we could move this function in fabioutils, which is IMO a better place, or uses to pkgsrc (an extra dependency) parser which was designed for this usecase.